File: android/src/main/java/org/ligi/gobandroid_hd/ui/fragments/NavigationAndCommentFragment.java
package org.ligi.gobandroid_hd.ui.fragments;

import android.os.Bundle;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.events.GameChangedEvent;
import org.ligi.gobandroid_hd.ui.go_terminology.GoTerminologyViewActivity;

public class NavigationAndCommentFragment extends GobandroidGameAwareFragment {


    private Handler gameChangeHandler = new Handler();
    private TextView myTextView;

    @Override
    public View createView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View res = inflater.inflate(R.layout.game_extra_review, container, false);

        myTextView = (TextView) res.findViewById(R.id.comments_textview);
        myTextView.setFocusable(false);

        res.findViewById(R.id.scrollview).setFocusable(false);

        getFragmentManager().beginTransaction().replace(R.id.container_for_nav, new NavigationFragment()).commit();
        onGoGameChanged(null);
        return res;
    }

    @Override
    public void onGoGameChanged(GameChangedEvent gameChangedEvent) {
        super.onGoGameChanged(gameChangedEvent);
        gameChangeHandler.post(new Runnable() {

            @Override
            public void run() {
                if (myTextView != null) {
                    myTextView.setText(game.getActMove().getComment());
                    GoTerminologyViewActivity.Companion.linkifyTextView(myTextView);
                }
            }

        });
    }

}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/fragments/NavigationAndCommentFragment.kt
package org.ligi.gobandroid_hd.ui.fragments

import android.os.Bundle
import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.comment_textview.*
import kotlinx.android.synthetic.main.comment_textview.view.*
import kotlinx.android.synthetic.main.game_extra_review.view.*
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.events.GameChangedEvent
import org.ligi.gobandroid_hd.ui.go_terminology.GoTerminologyViewActivity

class NavigationAndCommentFragment : GobandroidGameAwareFragment() {


    private val gameChangeHandler by lazy { Handler() }

    override fun createView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {

        val res = inflater.inflate(R.layout.game_extra_review, container, false)

        res.comments_textview.isFocusable = false

        res.scrollview.isFocusable = false

        onGoGameChanged(null)
        return res
    }

    override fun onGoGameChanged(gameChangedEvent: GameChangedEvent?) {
        super.onGoGameChanged(gameChangedEvent)
        gameChangeHandler.post {
            comments_textview?.let {
                it.text = game.actMove.comment
                GoTerminologyViewActivity.linkifyTextView(it)
            }
        }
    }

}
